openslides_backend/action/actions/user/forget_password.py

```diff
@@ -0,0 +1,136 @@
+
+    model = User()
+    schema = DefaultSchema(User()).get_default_schema(
+        title="user forget password schema",
+        additional_required_fields={"email": {"type": "string"}},
+    )
+    skip_archived_meeting_check = True
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        for instance in action_data:
+            email = instance.pop("email")
+
+            # check emailif valid email adress
+            if not self.check_email(EmailSettings.default_from_email):
+                raise ActionException(
+                    f"email '{EmailSettings.default_from_email}' is not a valid sender email address."
+                )
+            if not self.check_email(email):
+                raise ActionException(f"'{email}' is not a valid email adress.")
+
+            # search for users with email
+            filter_ = FilterOperator("email", "=", email)
+            results = self.datastore.filter(
+                self.model.collection, filter_, ["id", "username"]
+            )
+
+            # try to send the mail.
+            try:
+                with EmailMixin.get_mail_connection() as mail_client:
+                    for user in results.values():
+                        if self.sendmail(
+                            mail_client,
+                            email,
+                            self.get_email_subjext(),
+                            self.get_email_body(
+                                user["id"],
+                                self.get_token(user["id"], email),
+                                user["username"],
+                            ),
+                        ):
+                            yield {"id": user["id"], "last_email_send": round(time())}
+            except SMTPAuthenticationError as e:
+                raise ActionException(f"SMTPAuthenticationError: {str(e)}")
+            except SMTPSenderRefused as e:
+                raise ActionException(f"SMTPSenderRefused: {str(e)}")
+            except SSLCertVerificationError as e:
+                raise ActionException(f"SSLCertVerificationError: {str(e)}")
+            except Exception as e:
+                raise ActionException(
+                    f"Unspecified mail connection exception on sending invitation email to server {EmailSettings.host}, port {EmailSettings.port}: {str(e)}"
+                )
+
+    def get_token(self, user_id: int, email: str) -> str:
+        # get token without 'bearer '
+        return self.auth.create_authorization_token(user_id, email)[7:]
+
+    def get_email_body(self, user_id: int, token: str, username: str) -> str:
+
+        body_format = format_dict(
+            None,
+            {
+                "user_id": user_id,
+                "token": token,
+                "username": username,
+            },
+        )
+        return PW_FORGET_EMAIL_TEMPLATE.format_map(body_format)
+
+    def get_email_subjext(self) -> str:
+        return PW_FORGET_EMAIL_SUBJECT
```

Comment:
unnecessary function, you can just use the constant directly

Reply:
Removed.